File: GCD.py
#Function uses Euclid's formula to calculate GCD of two numbers.
def gcd(a,b):
    while(b!=0):
        if(a>b):
            a%=b
        else:
            temp=a
            a=b
            b=temp
    return a


num=input("Enter 2 numbers: ").split()
print("GCD of",num[0],"and",num[1],"is",gcd(int(num[0]),int(num[1])))

File: gcd.cpp
#include <bits/stdc++.h>
using namespace std;

// Function to return gcd of a and b
int gcd(int a, int b)
{
	while(b)
  {
    a = a % b;
    swap(a,b);
  }
  return a;
}

// Driver Code
int main()
{
	int a,b;
  cout<<"Enter the two numbers";
  cin>>a>>b;
	cout << "GCD(" << a << ", " << b << ") = "<<gcd(a, b)<< endl;
	return 0;
}

//code written by Trisha Moitra
